lib/index.js

```diff
@@ -194,27 +193,22 @@ Server.prototype.origins = function(v){
  */
 
 Server.prototype.listen =
-Server.prototype.attach = function(srv, opts){
-  if ('function' == typeof srv) {
+Server.prototype.attach = function (srv, opts) {
+  if ('function' === typeof srv) {
     var msg = 'You are trying to attach socket.io to an express ' +
     'request handler function. Please pass a http.Server instance.';
     throw new Error(msg);
   }
 
-  // handle a port as a string
-  if (Number(srv) == srv) {
-    srv = Number(srv);
-  }
-
-  if ('number' == typeof srv) {
+  // handle port as a string and number
+  if (!isNaN(srv)) {
     debug('creating http server and binding to %d', srv);
-    var port = srv;
-    srv = http.Server(function(req, res){
+    var port = Number(srv);
```

Comment:
Somewhat major refactoring here.

To avoid the use of `==`, I modified the handling of port number given as `string`; it is now handled together with port number given as `number` by checking if `srv` is not `NaN` (i.e. `!isNaN(srv)` on line 204)

Passed the test, but let me know if there is an issue with this